Language/Classes/Constructors/Factories/parameter_type_t01.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion For the dynamic semantics, assume that k is a redirecting factory
+/// constructor and k′ is the redirectee of k
+///
+/// It is a dynamic type error if an actual argument passed in an invocation of
+/// k is not a subtype of the actual type of the corresponding formal parameter
+/// in the declaration of k.
+///
+/// @description Checks that it is a dynamic type error if an actual argument
+/// type of a redirecting factory constructor call is not a subtype of the
+/// actual type of its corresponding formal parameter (that this situation can
+/// only occur when the static type of the given actual argument is `dynamic`).
```

Comment:
Oh, this is actually not precisely the situation we have below. We'd need to have different parameter types in the redirecting factory constructor and in the redirectee, because the whole point is that we satisfy one of the constraints and violate the other one. Something like this:

```dart
class A {
  factory A(int i) = B;
}

class B implements A {
  B(num n);
}

void main() {
  print('Call B');
  B(1.5 as dynamic); // OK.
  print('Tear off B then call');
  (B.new)(1.5 as dynamic); // OK.
  print('Call A');
  A(1.5 as dynamic); // Throws.
  print('Tear off A then call');
  (A.new)(1.5 as dynamic); // Throws.
}
```

This shows that the formal parameter types of the constructor declared by `A` are taken into account (that is, they aren't just ignored even though the invocation of `A(arg)` is very much the same thing as an invocation of `B(arg)`).

As far as I can see, we never have this distinction between the formal parameter types of the redirecting factory construct and the redirectee in this test.